### PR TITLE
[AUD-966] Add Collection Screen

### PR DESCRIPTION
**Pull request by sliptype**
### Description

* Adds
  * `CollectionScreen`
  * `CollectionScreenDetailsTile`
  * `SmartCollectionScreen`
  * Under The Radar smart collection screen
* Refactors `TrackScreenHeader` into `DetailsTile` which is used for tracks, collections, and smart collections. The main effort here was removing lots of branching at the leaf components in favor of configuration at the parent components
* Updates `useNavigation` to accept a type parameter to better model which params can be passed to which navigators
* Adds `start` prop to `Lineup`
* Moves collection state to common
* Uses InfoLabel pattern on collection page on web & mobile

Simulator Screen Shot - iPhone 13 - 2022-03-01 at 16 03 21
Simulator Screen Shot - iPhone 13 - 2022-03-01 at 16 04 05
Simulator Screen Shot - iPhone 13 - 2022-03-01 at 16 03 12

Left to do:
  * Wire up smart collection saving
  * Fix issue with pausing playback from TrackList
  * Fix issue with TrackList not playing on FavoritesScreen
  * Fix issue with scroll bar being in center of the screen

### Dragons

N/A, not being deployed yet

### How Has This Been Tested?

Manually on iOS simulator

### How will this change be monitored?

Will go through qa in the future

### CODE REVIEW

**Comment by sliptype** [comment]
Just noticed the InfoLabel details text doesn't seem to be vertically centered, will fix

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sk-aud-966-collection-screen

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sk-aud-966-collection-screen

File: packages/mobile/src/components/app-navigator/TopBar.tsx [hunk 1]
```diff
@@ -75,7 +76,7 @@ export const TopBar = memo(
     const styles = useStyles()
 
     const { neutralLight4 } = useThemeColors()
-    const navigation = useNavigation()
+    const navigation = useNavigation<MainParamList & SearchParamList>()
```
Comment (dylanjeffers) [inline packages/mobile/src/components/app-navigator/TopBar.tsx:79]: def a TODO for us is reexporting useNavigation with all the types built in

File: packages/mobile/src/components/details-tile/DetailsTile.tsx [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+    if (!user) {
+      return
+    }
+    navigation.push({
+      native: { screen: 'Profile', params: { handle: user.handle } },
+      web: { route: `/${user.handle}` }
+    })
+  }, [navigation, user])
+
+  const handleExternalLinkClick = useCallback(
```
Comment (dylanjeffers) [inline packages/mobile/src/components/details-tile/DetailsTile.tsx:212]: def fine for now, thinking we try and utilize a link hook or link component for some of this stuff

File: packages/web/src/pages/smart-collection/store/sagas.ts [hunk 3]
```diff
@@ -6,13 +6,17 @@ import { Track, UserTrack } from 'common/models/Track'
...
 import Explore from 'services/audius-backend/Explore'
 import { waitForBackendSetup } from 'store/backend/sagas'
 import { getLuckyTracks } from 'store/recommendation/sagas'
 import { EXPLORE_PAGE } from 'utils/route'
 import { requiresAccount, waitForValue } from 'utils/sagaHelpers'
 
+import {
+  fetchSmartCollection,
+  fetchSmartCollectionSucceeded
+} from '../../../common/store/pages/smart-collection/slice'
```
Comment (dylanjeffers) [inline packages/web/src/pages/smart-collection/store/sagas.ts:19]: same

**Review by dylanjeffers — APPROVED** [review]
massive lift, code all really clean, nice job

File: packages/mobile/src/components/app-navigator/TopBar.tsx (hunk 1 above)
Comment (sliptype) [reply packages/mobile/src/components/app-navigator/TopBar.tsx:79]: Well not all parameters will apply to all navigation contexts, that's why I made it a type parameter with `BaseStackParamList` as the default. That way we can usually avoid having to pass the type parameter but for certain navigation contexts like profile edit or search we can specify that param type

Comment (dylanjeffers) [reply packages/mobile/src/components/app-navigator/TopBar.tsx:79]: oh i see! doesnt navigate have the ability to hit any screen, even if it's outside that stack?

Comment (sliptype) [reply packages/mobile/src/components/app-navigator/TopBar.tsx:79]: I think that's only true if the containing navigator has been rendered: https://reactnavigation.org/docs/nesting-navigators/#navigating-to-a-screen-in-a-nested-navigator

Probably worth discussing synchronously though!

File: packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+  const tracksLineup = useSelectorWeb(getTracksLineup)
+  const tracksLoading = tracksLineup.status === Status.LOADING
+  const numTracks = tracksLineup.entries.length
+
+  const duration = tracksLineup.entries?.reduce(
+    (duration, entry) => duration + entry.duration,
+    0
+  )
+
+  const details =
```
[the remark this answers is not in the document]
Comment (sliptype) [reply packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx:77]: Yup! Good catch

File: packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+      : []
+
+  const isPlaying = useSelector(getPlaying)
+  const playingUid = useSelector(getPlayingUid)
+  const playingTrack = useSelector(getTrack)
+  const trackId = playingTrack?.trackId
+
+  const isQueued = tracksLineup.entries.some(entry => playingUid === entry.uid)
+
+  const recordPlay = (id, play = true) => {
```
[the remark this answers is not in the document]
Comment (sliptype) [reply packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx:99]: `recordPlay` is not a dependency for those useCallbacks, I can move this outside of the component

File: packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+    if (isPrivate) {
+      return messages.privatePlaylist
+    }
+
+    return messages.playlist
+  }, [isAlbum, isPrivate])
+
+  const renderTrackList = () => {
+    const trackMetadatas = tracksLineup.entries.map(track => ({
```
[the remark this answers is not in the document]
Comment (sliptype) [reply packages/mobile/src/screens/collection-screen/CollectionScreenDetailsTile.tsx:151]: Yeah this has to do with TrackList being ported from web where it worked like this. I agree would be nice to do another pass to make it easier to work with!

File: packages/mobile/src/screens/track-screen/TrackScreenDetailsTile.tsx [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+        id: track_id,
+        overflowActions
+      })
+    )
+  }
+
+  const renderHiddenHeader = () => {
+    return (
+      <View style={styles.hiddenDetailsTileWrapper}>
+        {/* <HiddeDetailsTile /> */}
```
[the remark this answers is not in the document]
Comment (sliptype) [reply packages/mobile/src/screens/track-screen/TrackScreenDetailsTile.tsx:289]: Still need to implement, just created an issue: https://linear.app/audius/issue/AUD-1636/hidden-track-screen-ui

File: packages/web/src/components/collection/mobile/CollectionHeader.js [hunk 8]
```diff
@@ -149,6 +149,33 @@ const CollectionHeader = ({
     [record]
   )
 
+  const collectionLabels = [
+    {
+      label: 'Tracks',
```
[the remark this answers is not in the document]
Comment (sliptype) [reply packages/web/src/components/collection/mobile/CollectionHeader.js:154]: Yeah gonna leave these for now, just touched these files to fix a styling issue Julian noticed

File: packages/web/src/components/collection/mobile/CollectionHeader.js [hunk 9]
```diff
@@ -149,6 +149,33 @@ const CollectionHeader = ({
...
+      value: formatDate(modified),
+      isHidden: variant === Variant.SMART
+    }
+  ].filter(({ isHidden, value }) => !isHidden && !!value)
+
+  const renderCollectionLabels = () => {
+    return collectionLabels.map(infoFact => {
+      return (
+        <div key={infoFact.label} className={styles.infoFact}>
+          <h2 className={styles.infoLabel}>{infoFact.label}</h2>
```
[the remark this answers is not in the document]
Comment (sliptype) [reply packages/web/src/components/collection/mobile/CollectionHeader.js:172]: Probably, but it's also like this in `InfoLabel.js`. I think we can clean this all up in the future

**Pull request merged**